spring-boot/src/main/java/org/springframework/boot/ResourceBanner.java

```diff
@@ -48,21 +52,25 @@
 
 	private static final Log logger = LogFactory.getLog(ResourceBanner.class);
 
-	private Resource resource;
+	private Resource bannerResource;
+	private Resource gitPropertiesResource;
+	private Resource buildInfoPropertiesResource;
+
+	public ResourceBanner(Resource bannerResource, Resource gitPropertiesResource, Resource buildInfoPropertiesResource) {
+		Assert.notNull(bannerResource, "Resource must not be null");
```

Comment:
The message should probably specify that the **banner** resource must not be null, since the other two resources can be null.